### ISSUE TITLE
Support showing MachineHealthChecks in `clusterctl describe cluster` output

### ISSUE THREAD

**Issue by ykakarap** [kubernetes-sigs/cluster-api#7222]
**User Story**

As a user I would like to be able see MachineHealthChecks acting on a cluster in the at-a-glance view generated by `clusterctl describe cluster`.

**Detailed Description**

Currently MHCs are not part of the clusterctl describe cluster output. Since MHCs can be uniquely linked to a specific cluster using `MachineHealthCheck.Spec.ClusterName` it would be nice to also see them in the output.

**Anything else you would like to add:**

They can be disabled by default and can sit behind a flag like `--show-machinehealthchecks`, similar to the other flags we currently have: `--show-templates`, `--show-resourcesets`, etc.

/kind feature
/area clusterctl

**Comment by fabriziopandini** [comment]
/triage accepted

it makes sense given that we are extending to all the objects in a cluster, but at the same time I'm not really sure what is the best way to represent them given the mechanism they are using to target machines

**Comment by ykakarap** [comment]
I guess this can be broken down into 2 things:
1) Listing the MachineHealthChecks that are linked to a cluster in `clusterctl describe`
2) Showing which MHCs are acting on which Machines

These things can probably be address in 2 steps. 
The first one is probably straight forward to implement with how `clusterctl describe` works right now. 
For the second one, it will need more though on a good UX to show that information. 

I think we can do 1 now and implement 2 later once we have a better idea on the UX.

**Comment by sbueringer** [comment]
+1 not sure if there's a good way to do 2. considering that MHC (as far as I know) could apply to Machines managed by different resources (KCP,MD)

**Comment by rajaSahil** [comment]
Hey @ykakarap, I would like to work on this.

**Comment by ykakarap** [comment]
@rajaSahil Great!
Feel free to assign it to yourself.

**Comment by rajaSahil** [comment]
Sure. Thank you!
/assign

**Comment by oscr** [comment]
Have you had any time to work on this @rajaSahil?

**Comment by fabriziopandini** [comment]
/unassign@rajaSahil

Wondering if what we need is clusterctl describe mhc, we rally neead to figure out what we want to achieve here given the relation betwen crs and clusters is 1:n

**Comment by oscr** [comment]
> 1. Listing the MachineHealthChecks that are linked to a cluster in `clusterctl describe`

If everyone agrees 1 is a good feature I would be willing to implement this.

**Comment by ykakarap** [comment]
> /unassign@rajaSahil
> 
> Wondering if what we need is clusterctl describe mhc, we rally neead to figure out what we want to achieve here given the relation betwen crs and clusters is 1:n

MHC to clusters is 1:1 right? MHC is uniquely matched to a Cluster using `mhc.spec.clusterName`. I guess usecase 1
> 1. Listing the MachineHealthChecks that are linked to a cluster in clusterctl describe
Is still a useful thing to have in the clusterctl describe output.

**Comment by sbueringer** [comment]
> MHC to clusters is 1:1 right?

Yes a MHC always belongs to a specific cluster, while a cluster can have multiple MHCs (which results in a 1:n relationship).

**Comment by fabriziopandini** [comment]
yeah sorry if I wasn't clear and a little bit imprecise, edited.

What I what thinking is a two-step process:
1. use `clusterctl describe cluster foo --show-mhc` to chek the list of mhc existing in a cluster (no further details provided)
2. use `clusterctl describe mhc foo-mhc1` to get a new view showing the machines targeted by this mhc instance, and might be also their health instead of the ready condition

but we can start from 1 for now

**Comment by oscr** [comment]
Thanks everyone for the clarifications! I'll give it a try during the weekend. 

/assign

**Comment by oscr** [comment]
I would like to ask for some advice how to look up the mhc(s). Would a viable option be via the [management cluster client](https://github.com/kubernetes-sigs/cluster-api/blame/efe3728d21683786bb7143ce6b2f57d68e0a18d5/cmd/clusterctl/client/describe.go#L65) to use [tree.Discovery()](https://github.com/kubernetes-sigs/cluster-api/blame/efe3728d21683786bb7143ce6b2f57d68e0a18d5/cmd/clusterctl/client/describe.go#L91)?

**Comment by killianmuldoon** [comment]
That sounds like the right way to approach this - i.e. insert a new `getMachineHealthChecksForCluster` somewhere in this function: https://github.com/kubernetes-sigs/cluster-api/blob/f2ba999757aa76b446943a9ffea14ecf8c7523e8/cmd/clusterctl/client/tree/discovery.go#L101
